catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/cql/MessageValidatorManagerTest.java

```diff
@@ -0,0 +1,164 @@
+    orderedListOfValidators.add(mockedValidatorOfNotTestSource);
+    orderedListOfValidators.add(firstMockedValidatorOfTestSource);
+    orderedListOfValidators.add(secondMockedValidatorOfTestSource);
+  }
+
+  @Test
+  public void testNullProcessingDetails() {
+    createManagerFromAndValidateMessagesOf(firstMockedValidatorOfTestSource, null);
+    assertThat(validWarningsFromRunOfThisTest, is(emptySet));
+  }
+
+  @Test
+  public void testProcessingDetailsWithNullSourceId() {
+    createManagerFromAndValidateMessagesOf(
+        firstMockedValidatorOfTestSource, mockedProcessingDetailsWithNullSourceId);
+    assertThat(validWarningsFromRunOfThisTest, is(emptySet));
+  }
+
+  @Test
+  public void testNullValidators() {
+    createManagerFromAndValidateMessagesOf(
+        (List<MessageValidator>) null, mockedProcessingDetailsFromTestSource);
+    assertThat(validWarningsFromRunOfThisTest, is(emptySet));
+  }
+
+  @Test
+  public void testEmptyValidators() {
+    createManagerFromAndValidateMessagesOf(
+        emptyListOfValidators, mockedProcessingDetailsFromTestSource);
+    assertThat(validWarningsFromRunOfThisTest, is(emptySet));
+  }
+
+  @Test
+  public void testValidatorsWithNullEntry() {
+    createManagerFromAndValidateMessagesOf(
+        listOfValidatorsWithNullEntry, mockedProcessingDetailsFromTestSource);
+    assertThat(validWarningsFromRunOfThisTest, is(emptySet));
+  }
+
+  @Test
+  public void testValidatorWithNullId() {
+    createManagerFromAndValidateMessagesOf(
+        mockedValidatorWithNullId, mockedProcessingDetailsFromTestSource);
+    assertThat(validWarningsFromRunOfThisTest, is(emptySet));
+  }
+
+  @Test
+  public void testProcessingDetailsWithUnsupportedSourceId() {
+    createManagerFromAndValidateMessagesOf(
+        mockedValidatorOfNotTestSource, mockedProcessingDetailsFromTestSource);
+    assertThat(validWarningsFromRunOfThisTest, is(emptySet));
+  }
+
+  @Test
+  public void testProcessingDetailsWithSupportedSourceId() {
+    createManagerFromAndValidateMessagesOf(
+        firstMockedValidatorOfTestSource, mockedProcessingDetailsFromTestSource);
+    assertThat(
+        validWarningsFromRunOfThisTest, is(validWarningsFromFirstMockedValidatorOfTestSource));
+  }
+
+  @Test
+  public void testMultipleValidators() {
+    createManagerFromAndValidateMessagesOf(
+        orderedListOfValidators, mockedProcessingDetailsFromTestSource);
+    assertThat(
+        validWarningsFromRunOfThisTest, is(validWarningsFromFirstMockedValidatorOfTestSource));
+  }
+
+  private void createManagerFromAndValidateMessagesOf(
+      MessageValidator validator, ProcessingDetails details) {
+    createManagerFromAndValidateMessagesOf(Collections.singletonList(validator), details);
+  }
+
+  private void createManagerFromAndValidateMessagesOf(
+      List<MessageValidator> validators, ProcessingDetails details) {
+    MessageValidatorManager messageValidatorManager = new MessageValidatorManager(validators);
```

Comment:
✏️ Consider making this method return a set of warnings rather than using the class scoped variable